barretenberg/cpp/pil/avm/gadgets/mem_slice.pil

```diff
@@ -0,0 +1,39 @@
+include "../main.pil";
+
+namespace slice(256);
```

Comment:
I think I get what this does, my only comment is naming:
calling it `slice` (or better, `mem_slice`) makes it look like it's generic but then even the core relations like "addresses should be decreasing" and "cnt management" do use calldatacopy-specific selectors. Should this just be called the calldata gadget?

(I do understand that even if the gadget was a generic slice read/write gadget then there would be _some_ mention of calldata, just like it happens with normal memory... we need at least some selectors for lookups/perms)

Reply:
What will happen for sure is that return opcode will be performed in the same trace. So, there will be specific selectors for calldatacopy and return opcode. Other columns like cnt, and addr, can be re-used. That is why I named it slice. I chose a short prefix, as the namespace is prefixed for any field of this trace.